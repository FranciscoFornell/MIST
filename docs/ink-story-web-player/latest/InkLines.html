<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
<<<<<<< Updated upstream
  <title>Docs: Simple Ink Story Web Player InkLines</title>
=======
  <title>Docs: Ink Story Web Player InkLines</title>
>>>>>>> Stashed changes

  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  <script src="./build/entry.js"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,700|Inconsolata,700" rel="stylesheet">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
  <link type="text/css" rel="stylesheet" href="styles/app.min.css">
  <link type="text/css" rel="stylesheet" href="styles/iframe.css">
  <link type="text/css" rel="stylesheet" href="">
  <script async defer src="https://buttons.github.io/buttons.js"></script>

  
</head>



<body class="layout small-header">
    <div id="stickyNavbarOverlay"></div>
    

<div class="top-nav">
    <div class="inner">
        <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
        </a>
        <div class="logo">
            
             
                <a href="index.html">
<<<<<<< Updated upstream
                    <h1 class="navbar-item">Simple Ink Story Web Player</h1>
=======
                    <h1 class="navbar-item">Ink Story Web Player</h1>
>>>>>>> Stashed changes
                </a>
            
        </div>
        <div class="menu">
            
            <div class="navigation">
                <a
                    href="index.html"
                    class="link"
                >
                    API Documentation
                </a>
                
                 
                    
                        <a
                            class="link user-link "
                            href="https://github.com/FranciscoFornell/MIST/tree/master/packages/ink-story-web-player"
                        >
                            Github
                        </a>
                    
                
                
            </div>
        </div>
    </div>
</div>
    <div id="main">
        <div
            class="sidebar "
            id="sidebarNav"
        >
            
            <nav>
                
                    <h2><a href="index.html">Documentation</a></h2><div class="category"><h3>Components</h3><ul><li><a href="InkButton.html">InkButton</a></li><li><a href="InkChoices.html">InkChoices</a></li><li><a href="InkHeader.html">InkHeader</a></li><li><a href="InkLines.html">InkLines</a></li><li><a href="InkStoryWebPlayer.html">InkStoryWebPlayer</a></li></ul></div>
                
            </nav>
        </div>
        <div class="core" id="main-content-wrapper">
            <div class="content">
                <header class="page-title">
                    <p>Components</p>
                    <h1>InkLines</h1>
                </header>
                




<section>

<header>
    
        <h2>
            &ltInkLines <i>lines</i> /&gt
        </h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class='vertical-section'>
                <div class="members">
                    <div class="member">
                        <div class=name>
                            <span class="tag">Constructor</span>
                        </div>
                        


    
    <h4 class="name" id="InkLines">
        <a class="href-link" href="#InkLines">#</a>
        
        <span class="code-name">
            
                &ltInkLines <i>lines</i> /&gt
            
        </span>
    </h4>
    

    
    
    <div class="description">
        Simple Ink Web Player lines component
    </div>
    







    
        <h5>PropTypes:</h5>
        
<div class="table-container">
  <table class="params table">
    <thead>
    <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Required</th>
        <th>Description</th>
        <th class="last">Default</th>
    </tr>
    </thead>

    <tbody>
    
          <tr class="deep-level-0">
            <td class="name"><code>lines</code></td>
            <td>
              arrayOf
            </td>
            <td>Yes</td>
            <td>Array of lines</td>
            <td></td>
          </tr>
          
    
    </tbody>
  </table>
</div>
    







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="InkLines_index.js.html" class="button">View Source</a>
            <span>
                <a href="InkLines_index.js.html">InkLines/index.js</a>, <a href="InkLines_index.js.html#line20">line 20</a>
            </span>
        </p>
    
</dl>





















    <h5>Example</h5>
    
    
        

<div id='component0'></div>

<script>
  ReactDOM.render(ReactWrapper({
    example: "const lines = [\n  'Line 1',\n  'Line 2',\n  'Line 3',\n  'Line 4',\n  'Line 5',\n  'Line 6',\n  'Line 7',\n  'Line 8',\n];\nreturn (<InkLines lines={lines} />);",
    uniqId: "component0",
  }), document.getElementById('component0'));
</script>
    



                    </div>
                </div>
            </div>
        
    
    </div>
    
        
    
    

    

    

    

    

    

    

    

    

    
</article>

</section>




            </div>
            
            <footer class="footer">
                <div class="content has-text-centered">
                    <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.6</a></p>
                    <p class="sidebar-created-by">
                        <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
                        <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
                    </p>
                </div>
            </footer>
            
        </div>
        <div id="side-nav" class="side-nav">
        </div>
    </div>
<script src="scripts/app.min.js"></script>
<script>PR.prettyPrint();</script>
<script src="scripts/linenumber.js"> </script>

</body>
</html>