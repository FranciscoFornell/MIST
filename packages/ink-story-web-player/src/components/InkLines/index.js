import React from 'react';
import PropTypes from 'prop-types';

/**
 * Simple Ink Web Player lines component
 * @component
 * @example
 * const lines = [
 *   'Line 1',
 *   'Line 2',
 *   'Line 3',
 *   'Line 4',
 *   'Line 5',
 *   'Line 6',
 *   'Line 7',
 *   'Line 8',
 * ];
 * return (<InkLines lines={lines} />);
 */
const InkLines = ({ lines }) => {
  let counter = 0;
  return (
    <React.Fragment>
      {lines.map((line) => (
        <p key={counter++}>{line}</p>
      ))}
    </React.Fragment>
  );
};

InkLines.propTypes = {
  /**
   * Array of lines
   */
  lines: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default InkLines;
