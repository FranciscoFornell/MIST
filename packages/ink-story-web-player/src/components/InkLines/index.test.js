import React from 'react';
import { render } from '@testing-library/react';
import InkLines from './index';

let lines;

const renderSnapshot = () =>
  render(<InkLines lines={lines} />).container.innerHTML;

afterEach(() => {
  jest.clearAllMocks();
  lines = [];
});
afterAll(() => {
  jest.restoreAllMocks();
});

describe('InkLines component', () => {
  it('should render no lines if an ampty lines array is given', () => {
    lines = [];
    expect(renderSnapshot()).toMatchSnapshot();
  });
  it('should render the lines correctly', () => {
    lines = ['Line 1', 'Line 2', 'Line 3'];
    expect(renderSnapshot()).toMatchSnapshot();
  });
});
