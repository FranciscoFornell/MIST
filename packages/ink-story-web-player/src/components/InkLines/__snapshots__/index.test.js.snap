// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InkLines component should render no lines if an ampty lines array is given 1`] = `""`;

exports[`InkLines component should render the lines correctly 1`] = `
<p>
  Line 1
</p>
<p>
  Line 2
</p>
<p>
  Line 3
</p>
`;
